### ISSUE TITLE
Label compatibility matching does not seem to work correctly

### ISSUE THREAD

**Issue by dgeelen** [hudson3-plugins/accelerated-build-now-plugin#2]
Hi. I'm trying to use the Accelerated Build Now plugin, since we have a lot of jobs that are always queued, but sometimes I need to build a specific job NOW.

The problem is that the plugin appears to do nothing at all. Looking at the Jenkins log file, I see this:

```
Jul 03, 2015 3:00:41 PM INFO org.terracotta.jenkins.plugins.acceleratedbuildnow.AcceleratedBuildNowAction doBuild
project : ProjectXXX needs to be built NOW !
Jul 03, 2015 3:00:41 PM INFO org.terracotta.jenkins.plugins.acceleratedbuildnow.AcceleratedBuildNowAction doBuild
project : ProjectXXX is scheduled to build now !
Jul 03, 2015 3:00:41 PM INFO org.terracotta.jenkins.plugins.acceleratedbuildnow.AcceleratedBuildNowAction slaveRunningBuildCompatible
build : 501 of project ProjectYYY is not running on node compatible with Git&&VisualStudio
Jul 03, 2015 3:00:41 PM INFO org.terracotta.jenkins.plugins.acceleratedbuildnow.AcceleratedBuildNowAction doBuild
project : ProjectXXX could not be built : no way to build it now !
```

However, looking at my slave specifications, it has many labels, including VisualStudio and Git. In fact, if it did not have the VisualStudio and Git labels, it could never have started to build ProjectYYY there.

**Comment by bobfoster** [comment]
If you're looking at the Jenkins log file, perhaps this is directed to the
wrong place?

**Comment by dgeelen** [comment]
No, this is output from the Accelerated Build Now! plugin, it just happens to end up in the jenkins log.

The code generating the incompatibility message is here:

``` java
  private boolean slaveRunningBuildCompatible(AbstractBuild lastBuild, Label assignedLabel) {
    boolean contains = assignedLabel == null ? true : lastBuild.getBuiltOn().getAssignedLabels().contains(assignedLabel);
    if (!contains) {
      LOG.info("build : " + lastBuild.getNumber() + " of project " + lastBuild.getProject().getName() + " is not running on node compatible with " + assignedLabel.getName());
    }
    return contains;
  }
```

See https://github.com/hudson3-plugins/accelerated-build-now-plugin/blob/master/src/main/java/org/terracotta/jenkins/plugins/acceleratedbuildnow/AcceleratedBuildNowAction.java , ~line 200

**Comment by rogierschouten** [comment]
+1

**Comment by dgeelen** [comment]
Hi bob,

Could you please give some feedback on this issue? I still feel this is a bug in the plugin, it doesn't seem to understand the label expression syntax. E.g. when you have a label like 'A&&B', this means that the slave offers both features A and B. So if a build only needs A, it can run on that slave. However I think the plugin just looks at the strings "A" and "A&&B", and decides that they're different, and that hence the build cannot run on that slave.

What do you think?

**Comment by bobfoster** [comment]
Sorry for the ambiguous message. I meant that Hudson has no Jenkins log. I
see you linked to the Hudson doc, so I'll assume you are running Hudson and
not Jenkins.

I haven't had time to look at the plugin. I wonder if Anthony Dahanne
https://github.com/anthonydahanne has any insight into the issue?

Bob
